sql/core/src/main/scala/org/apache/spark/sql/jdbc/PostgresDialect.scala

```diff
@@ -100,5 +100,5 @@ private object PostgresDialect extends JdbcDialect {
 
   }
 
-  override def isCascadingTruncateTable(): Option[Boolean] = Some(true)
+  override def isCascadingTruncateTable(): Option[Boolean] = Some(false)
```

Comment:
cc @dongjoon-hyun https://github.com/apache/spark/pull/14086

Reply:
Actually @gatorsmile made the comment in #14086 that Postgres cascades truncates by default. Pointing to the exact same documentation as I do in the JIRA, but then referring to the opposite behavior. I still stand by my Jira and included proof (see Jira). Also Postgres supports a rollback of a truncate.

Reply:
Sorry, you are the author of that comment. Missed that :)

Reply:
Thank you for pining me, @gatorsmile .

Reply:
@bolkedebruin .

Originally, Spark uses `DROP TABLE` for Overwrite mode. #14086 introduced `TRUNCATE` operation when `JdbcUtils.isCascadingTruncateTable(url) == Some(false)`. 

So, for PostgresDialect, Spark doesn't use TRUNCATE here.

Reply:
@dongjoon-hyun indeed, Spark does not use `TRUNCATE` for Postgres, due to the value of `JdbcUtils.isCascadingTruncateTable(url)`. The result, however, is that it will try to drop the table, which will fail for the exact same reason if there are tables dependent on the table to be truncated/dropped (e.g. a foreign key constraint). I think this raises another issue, that it should be possible to specify a `cascade` flag for these operations (but that's for another JIRA) Moreover, the definition of the `JdbcUtils.isCascadingTruncateTable` function is: ```Return Some[true] iff `TRUNCATE TABLE` causes cascading default.```, which isn't the case for Postgres (as mentioned in the JIRA)

Reply:
Previously, we referred the following sentence in https://www.postgresql.org/docs/current/static/sql-truncate.html.

```
name

    The name (optionally schema-qualified) of a table to truncate. If ONLY is specified before the table name, only that table is truncated. If ONLY is not specified, the table and all its descendant tables (if any) are truncated. Optionally, * can be specified after the table name to explicitly indicate that descendant tables are included.
```

For your test case, it seems to be different, but could you point us the sentence you're referring exactly?

Reply:
```

CASCADE

    Automatically truncate all tables that have foreign-key references to any of the named tables, or to any tables added to the group due to CASCADE.

``` 
So I think the difference is that `CASCADE` will also affect tables with foreign-key references to the table in question, rather than only its descendant tables. The function definition (and its name) seems to point more towards the definition of `CASCADE` behaviour than `ONLY` imho

Reply:
@danielvdende .

Actually, I proposed TRUNCATE version as an optimized version of this operation.
If DROP TABLE failed with the same reasons, it seems to be legitimate for some reasons (your configurations or previleges)

BTW, if you enables this, does it solve your situation?

Reply:
Spark don't use `CASCADE` at all. I'm wondering if your PR will not solve your problem.
```scala
  def truncateTable(conn: Connection, table: String): Unit = {
    val statement = conn.createStatement
    try {
      statement.executeUpdate(s"TRUNCATE TABLE $table")
    } finally {
      statement.close()
    }
  }
```

Reply:
I've looked at your test case again. `fix` is not a decendant table of `info`, it's a parent table.
```
domino=> truncate domino.info;
ERROR:  cannot truncate a table referenced in a foreign key constraint
DETAIL:  Table "fix" references "info".
HINT:  Truncate table "fix" at the same time, or use TRUNCATE ... CASCADE.
```

Reply:
@dongjoon-hyun to be clear, I think there are 2 problems:
1. The PostgresDialect indicates that `CASCADE` is enabled by default for Postgres. This isn't the case as the Postgres docs show. 
2. As you correctly mention (this is what in my previous comment), Spark doesn't use `CASCADE` at all, which, especially considering the method this PR edits, is a bit odd I think. I plan to open a different JIRA ticket for this, and add it. This will be more work, and is outside the scope of the current JIRA.

Reply:
```
airflow=# CREATE TABLE products (
airflow(#     product_no integer PRIMARY KEY,
airflow(#     name text,
airflow(#     price numeric
airflow(# );
CREATE TABLE
airflow=#
airflow=# CREATE TABLE orders (
airflow(#     order_id integer PRIMARY KEY,
airflow(#     product_no integer REFERENCES products (product_no),
airflow(#     quantity integer
airflow(# );
CREATE TABLE

airflow=# insert into products VALUES (1, 1, 1);
INSERT 0 1
airflow=# insert into orders VALUES (1,1,1);
INSERT 0 1
airflow=# select * from products;
 product_no | name | price
------------+------+-------
          1 | 1    |     1
(1 row)

airflow=# select * from orders;
 order_id | product_no | quantity
----------+------------+----------
        1 |          1 |        1
(1 row)

airflow=# truncate orders;
TRUNCATE TABLE
airflow=# select * from products;
 product_no | name | price
------------+------+-------
          1 | 1    |     1
(1 row)
```

Reply:
Please note that drop/create is an expensive operation. In addition I don't think (imho) spark should ever do a drop/create as it changes the schema.

Reply:
Thanks for the extended example. The following was the original motivation of #14086 . I'm not disagree with you on this.
```
`drop/create` is an expensive operation
```

Reply:
In not sure if you are agreeing or not agreeing with me now :-) , but at least as just shown truncate is supported and does not cascade by default on Postgres. Can we conclude that this change seems right for Postgres, which is the only affected supported database by Spark - others default to truncate?